### DISCUSSION TITLE
Follow target on one axis only?

### DISCUSSION THREAD

**Discussion by JamesLoyMartin** [ramokz/phantom-camera#421] [Question / Help Wanted] [answered]
First, I love Phantom Camera. Thanks for making it!

So is there a way to follow the follow_target along only one axis and not both x and y?
I'm trying to follow the player only along the x axis at certain times.

Specifically, I want the camera to 'platform snap', meaning the camera does not move on the Y axis when the character is jumping in the air, but only when he is on the ground. Used in Super Mario World and Donkey Kong Country 2 onwards.

Here is an example of how i achieved platform snapping with phantom camera so far:
![platformSnapExamp](https://github.com/user-attachments/assets/d5c75b3b-08c6-4d4d-824b-01191933c429)
you can see the camera follow_target at the top.

My solution was to have the phantom camera follow a different target (the triangle at the player's feet) when the player jumps, then switch the follow_target back to the player when he lands. 

It looks nice in the GIF, but really there is a noticeable one frame pause when switching the follow_target back and forth as the other target following the player lags behind one physics frame. (I'm simply doing position.x = player.position.x in physics process for lack of a better way to follow the player, but it lags a frame.)

Ideally, I'd like to just always follow the player and disable following along the y axis instead of switching follow_target to avoid the lag. Is that possible or is there another approach you would recommend?

Thank you!

**Comment by ramokz** [comment, marked as answer]
There isn't a non-DIY solution for this atm, but think adding proper support for it is definitely worthwhile.

The idea was also brought up in a discussion about adding support for Look Ahead; where the camera will move further ahead in the direction the target is moving in. Don't think locking an axis will introduce a lot of work on its own, so can see it being officially supported sooner than that.

Feel free to write a [feature proposal](https://github.com/ramokz/phantom-camera/issues/new?assignees=&labels=feature-proposal%2Cneeds-triage&projects=&template=feature_proposal.yml) as that would help track the popularity of it. Am otherwise working out from my priority list, so always appreciate others sharing where they feel gaps exist :)

**Reply by JamesLoyMartin** [reply]
Cool, thanks good to know!
I'll make a proposal.

I'll also just mention for what its worth, that look ahead works really solid by just tweening the follow_offset of the phantom camera. Perfectly smooth, no issues. 

Here's a preview of it:
![PcamOffsetPreview](https://github.com/user-attachments/assets/c6590a93-f190-4168-9bca-3905e78a5b37)
(actual float values are rounded in label)
